src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/InductionScheduleUpdateEventPusher.kt

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import uk.gov.justice.hmpps.sqs.HmppsTopic
+import uk.gov.justice.hmpps.sqs.publish
+import java.net.URI
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+
+@Component
+class InductionScheduleUpdateEventPusher(
+  private val hmppsQueueService: HmppsQueueService,
+  private val objectMapper: ObjectMapper,
+  @Value("\${service.base-url}") private val serviceBaseUrl: String,
+) {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+    const val EVENT_TYPE = "plp.induction-schedule.updated"
+    const val CONTENT_TYPE = "text/plain;charset=UTF-8"
+  }
+
+  internal val eventTopic by lazy { hmppsQueueService.findByTopicId("domainevents") as HmppsTopic }
+
+  fun sendEvent(prisonerNumber: String, occurredAt: Instant) {
+    val cne = HmppsDomainEvent(
```

Comment:
`cne` ? - could we call it `domainEvent` or similar?

Reply:
ha! a lot of this has been cribbed from other projects (cne = case note event Ooops!)